File: src/config/appConfig.js
/**
 * Centralized configuration for the Movie Club web app
 * This file contains passwords, admin settings, and other configuration values
 */

const AppConfig = {
  // MARK: - Admin Configuration
  
  /**
   * Email addresses that are automatically granted admin privileges
   * (For future authentication implementation)
   */
  adminEmails: new Set([
    "rattanajoey@gmail.com"
  ]),
  
  /**
   * Password required to access the admin panel
   */
  adminPanelPassword: "adminpass",
  
  /**
   * Password required to publish monthly selections
   */
  publishSelectionsPassword: "thunderbolts",
  
  // MARK: - Oscar Voting Configuration
  
  /**
   * Password required to access Oscar voting
   */
  oscarVotingPassword: "oscar2025",
  
  /**
   * Whether Oscar voting is currently enabled
   */
  oscarVotingEnabled: true,
  
  // MARK: - Submission Configuration
  
  /**
   * Password required to submit movies
   */
  movieSubmissionPassword: "thunderbolts",
  
  // MARK: - Feature Flags
  
  /**
   * Whether submissions are currently open
   */
  submissionsOpen: false,
  
  /**
   * Whether to show the submission list
   */
  showSubmissionList: true,
  
  /**
   * Whether to show the holding pool
   */
  showHoldingPool: true,
  
  // MARK: - Firebase Collections
  
  collections: {
    monthlySelections: "MonthlySelections",
    genrePools: "GenrePools",
    submissions: "Submissions",
    holdingPool: "HoldingPool",
    oscarCategories: "OscarCategories",
    oscarVotes: "OscarVotes",
    users: "users"
  },
  
  // MARK: - Helper Functions
  
  /**
   * Check if an email should have admin privileges
   * @param {string} email - The email to check
   * @returns {boolean} - True if email is in admin list
   */
  isAdminEmail: function(email) {
    if (!email) return false;
    return this.adminEmails.has(email.toLowerCase());
  },
  
  /**
   * Get password from environment variables or use default
   * @param {string} envKey - Environment variable key
   * @param {string} defaultValue - Default value if env var not found
   * @returns {string} - The password to use
   */
  getPassword: function(envKey, defaultValue) {
    return process.env[envKey] || defaultValue;
  }
};

export default AppConfig;

/**
 * Usage Examples:
 * 
 * 1. Import the config:
 *    import AppConfig from '../config/appConfig';
 * 
 * 2. Check admin access:
 *    if (password === AppConfig.adminPanelPassword) {
 *      // Grant access
 *    }
 * 
 * 3. Check if email is admin:
 *    if (AppConfig.isAdminEmail(userEmail)) {
 *      // Auto-grant admin privileges
 *    }
 * 
 * 4. Use feature flags:
 *    if (AppConfig.oscarVotingEnabled) {
 *      // Show Oscar voting button
 *    }
 * 
 * 5. Access collection names:
 *    collection(db, AppConfig.collections.monthlySelections)
 * 
 * 6. Override with environment variables:
 *    - Add to .env file: REACT_APP_ADMIN_PASSWORD=your_secure_password
 *    - Use: AppConfig.getPassword('REACT_APP_ADMIN_PASSWORD', AppConfig.adminPanelPassword)
 */

File: Movie Club Cafe/Movie Club Cafe/Config/AppConfig.swift
//
//  AppConfig.swift
//  Movie Club Cafe
//
//  Created by Kavy Rattana on 11/6/25.
//

import Foundation

/// Centralized configuration for the Movie Club app
/// This file contains passwords, API keys, and other configuration values
struct AppConfig {
    
    // MARK: - Admin Configuration
    
    /// Email addresses that are automatically granted admin privileges
    static let adminEmails: Set<String> = [
        "rattanajoey@gmail.com"
    ]
    
    /// Password required to access the admin panel
    static let adminPanelPassword = "adminpass"
    
    /// Password required to publish monthly selections
    static let publishSelectionsPassword = "thunderbolts"
    
    /// Check if an email should have admin privileges
    static func isAdminEmail(_ email: String?) -> Bool {
        guard let email = email?.lowercased() else { return false }
        return adminEmails.contains(email)
    }
    
    // MARK: - Oscar Voting Configuration
    
    /// Password required to access Oscar voting
    static let oscarVotingPassword = "oscar2025"
    
    /// Whether Oscar voting is currently enabled
    static let oscarVotingEnabled = true // Set to false during off-season
    
    // MARK: - Submission Configuration
    
    /// Password required to submit movies (matches web app)
    static let movieSubmissionPassword = "thunderbolts"
    
    // MARK: - Feature Flags
    
    /// Whether submissions are currently open
    static let submissionsOpen = false
    
    /// Whether to show the submission list
    static let showSubmissionList = true
    
    /// Whether to show the holding pool
    static let showHoldingPool = true
    
    // MARK: - Firebase Collections
    
    struct Collections {
        static let monthlySelections = "MonthlySelections"
        static let genrePools = "GenrePools"
        static let submissions = "Submissions"
        static let holdingPool = "HoldingPool"
        static let oscarCategories = "OscarCategories"
        static let oscarVotes = "OscarVotes"
        static let users = "users"
    }
    
    // MARK: - Environment-based Configuration
    
    /// Get configuration value from environment or use default
    static func getPassword(key: String, defaultValue: String) -> String {
        if let envValue = ProcessInfo.processInfo.environment[key] {
            return envValue
        }
        return defaultValue
    }
    
    // MARK: - Helper Functions
    
    /// Get the admin password from environment or default
    static var adminPassword: String {
        getPassword(key: "ADMIN_PANEL_PASSWORD", defaultValue: adminPanelPassword)
    }
    
    /// Get the publish password from environment or default
    static var publishPassword: String {
        getPassword(key: "PUBLISH_PASSWORD", defaultValue: publishSelectionsPassword)
    }
    
    /// Get the Oscar voting password from environment or default
    static var oscarPassword: String {
        getPassword(key: "OSCAR_PASSWORD", defaultValue: oscarVotingPassword)
    }
}

// MARK: - Usage Examples

/*
 
 To use these configurations in your code:
 
 1. Import the config:
    import Config (if needed)
 
 2. Access passwords:
    if password == AppConfig.adminPassword {
        // Grant access
    }
 
 3. Use feature flags:
    if AppConfig.oscarVotingEnabled {
        // Show Oscar voting button
    }
 
 4. Access collection names:
    db.collection(AppConfig.Collections.monthlySelections)
 
 5. Override with environment variables:
    In Xcode:
    - Edit Scheme → Run → Arguments → Environment Variables
    - Add: ADMIN_PANEL_PASSWORD = your_secure_password
 
 */

